config/opal_check_pc.m4

```diff
@@ -0,0 +1,109 @@
+          which pkg-config > /dev/null 2>&1
+          AS_IF([test "$?" -eq "0"],
+                [
+                  AC_MSG_RESULT([yes])
+                  OPAL_HAS_PKG_CONFIG="yes"
+                ],
+                [
+                  AC_MSG_RESULT([no])
+                  OPAL_HAS_PKG_CONFIG="no"
+                  OPAL_FLAGS_APPEND_UNIQ([OMPI_WRAPPER_EXTRA_LIBS], [$OPAL_EXTRA_LIBS])
+                ]
+          )
+        ]
+  )
+])
+
+dnl OPAL_CHECK_LFLAGS
+dnl Get the -l flags using pkg-config on the passed in .pc file, and
+dnl append them to OMPI_WRAPPER_EXTRA_LIBS.
+dnl
+dnl Requires 1 or more arguments.
+dnl
+dnl Arg 1 (Required): the name of the .pc file to look for (hwloc/pmix/libevent_core).pc
+dnl 
+dnl Arg 2..n (Optional): The search path for the .pc file above and any dependencies.
+dnl                      These search will be appended and set in the env variable
+dnl                      PKG_CONFIG_PATH to tell pkg-config where to locate the .pc
+dnl                      file from argument 1, and its dependencies.
+dnl
+AC_DEFUN([OPAL_CHECK_LFLAGS], [
+
+  OPAL_CHECK_PKG_CONFIG
+  AS_IF([test "$OPAL_HAS_PKG_CONFIG" = "yes"],
+        [
+          AC_MSG_NOTICE([Looking for pc file for $1])
+          OPAL_VAR_SCOPE_PUSH([opal_pkg_config_name opal_pkg_config_env_to_set opal_pkg_config_args opal_pkg_config_cmd opal_pkg_config_result])
+          opal_pkg_config_name=$1
+          opal_pkg_config_env_to_set=""
+          opal_pkg_config_cmd=""
+
+          AS_LITERAL_WORD_IF([$1], [], [m4_fatal([Non-literal argument $1])], [])
+          $1_OMPI_PC_DIR=""
+
+          # Tell pkg-config where to find the .pc file from argument 1
+          # as well as any/all dependencies.
+          # The following block will append the path
+          # to those .pc files into the pkg-config command.
+          # num_paths > 1 implies there is at least one dependent .pc file
+          # to append.
+          AS_IF([test $# -gt 1],
+                [
+                  $1_OMPI_PC_DIR=$2
+
+                  # Shift the arguments by one to get to the actual paths.
+                  opal_pkg_config_args=m4_shift($@)
+
+                  # Iterate over the comma seperated arguments, and replace the ','
+                  for i in $(echo $opal_pkg_config_args | sed "s/,/ /g"); do
+                    opal_pkg_config_env_to_set="$i:${opal_pkg_config_env_to_set}"
+                  done
+                  opal_pkg_config_cmd="env PKG_CONFIG_PATH=$opal_pkg_config_env_to_set pkg-config --libs $opal_pkg_config_name"
+                ],
+                [opal_pkg_config_cmd="pkg-config --libs $opal_pkg_config_name"]
+          )
+
+          OPAL_LOG_MSG([pkg-config cmd for $opal_pkg_config_name: $opal_pkg_config_cmd])
+
+          opal_pkg_config_result=$($opal_pkg_config_cmd)
+          AS_IF([test -z "$opal_pkg_config_result"],
+                [
+                  AC_MSG_WARN([Could not find viable $opal_pkg_config_name.pc])
```

Comment:
@bwbarrett for some reason libevent_core/libevent_pthreads are not located in the default location on aws ci for libevent external builds.

Not sure if they don't exist, or just don't exist where pkg-config looks for them. To get it to pass I changed this from a hard error to a warning.

Reply:
I'm pretty sure the external libevent is installed in a user directory (rather than a system directory), which is why pkg-config would be wonky.  This probably isn't an unusual config, so I think your change is the right one.  I'll think about the right way to add some tests for this  (as well as external PMIx/PRRTE) in the coming weeks.